vault/audited_headers.go

```diff
@@ -120,8 +131,15 @@ func (c *Core) setupAuditedHeadersConfig() error {
 		}
 	}
 
+	// Ensure that we are able to case-sensitively access the headers;
```

Comment:
On a side note, I wonder if we should mention mandatory upgrade to specific versions when deprecating any of the upgrade code, in general.